### PR TITLE
Add system tests

### PR DESCRIPTION
**Pull request by BeardedWhale**

### CODE REVIEW

File: .tests/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ENV HTTP_PROXY_ADDRESS=$http_url
+ENV GRPC_PROXY_ADDRESS=$grpc_url
+ENV SECURE=$grpc_secure
+
+ENV AWS_STORAGE_ENDPOINT=$aws_storage_endpoint
+ENV AWS_ACCESS_KEY_ID=$aws_access_key_id
+ENV AWS_SECRET_ACCESS_KEY=$aws_secret_access_key
+
+RUN hs cluster add --name $CLUSTER_NAME --server $HTTP_PROXY_ADDRESS
+RUN hs cluster use $CLUSTER_NAME
```
Comment (akastav) [inline .tests/Dockerfile:32]: in the process of building the container, you immediately generate a config for hs and there will always be localhost: 80, if you run such a container, it is useless to redefine env. 
So I added buildargs. 
If you want it to work, you need to create an entrypoint and generate the configuration when starting the container

File: .tests/models/adult/model/serving_test.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  outputs:
+    classes:
+      profile: numerical
+      shape: scalar
+      type: int64
+install-command: pip install -r requirements.txt
+kind: Model
+monitoring:
+- config:
+    monitoring-model: adult_monitoring_test:27
```
Comment (tidylobster) [inline .tests/models/adult/model/serving_test.yaml:62]: Specifying monitoring config here to a specific model version can cause tests to fail if some previous tests were failed. Can we omit it here?

File: .tests/models/adult/model/src/func_main.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            'hours_per_week',
+            'country']
+
+
+def extract_value(proto):
+    return np.array(proto.int64_val, dtype='int64')[0]
+
+
+def predict(**kwargs):
+    # extracted = np.array([extract_value(kwargs[feature]) for feature in features])
```
Comment (tidylobster) [inline .tests/models/adult/model/src/func_main.py:26]: Remove unnecessary comments here, they are no longer relevant.

File: .tests/models/adult/serving.yaml [hunk 4]
```diff
@@ -0,0 +1,146 @@
+kind: Model
+name: "adult_monitoring"
+payload:
+  - "monitoring_model/src/"
+  - "monitoring_model/requirements.txt"
+  - "monitoring_model/monitoring_model.joblib"
+runtime: "hydrosphere/serving-runtime-python-3.6:0.1.2-rc0"
```
Comment (tidylobster) [inline .tests/models/adult/serving.yaml:7]: An old version of the runtime.

File: .tests/models/adult/serving.yaml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  model: adult_monitoring:1
+
+---
+kind: Model
+name: "adult_classification"
+payload:
+  - "model/src/"
+  - "model/requirements.txt"
+  - "model/classification_model.joblib"
+runtime: "hydrosphere/serving-runtime-python-3.6:0.1.2-rc0"
```
Comment (tidylobster) [inline .tests/models/adult/serving.yaml:83]: An old version of the runtime.

File: .tests/models/adult/serving.yaml [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  outputs:
+    classes:
+      shape: scalar
+      type: int64
+      profile: numerical
+---
+kind: Application
+name: "adult_classification"
+singular:
+  model: adult_classification:1
```
Comment (tidylobster) [inline .tests/models/adult/serving.yaml:146]: Not sure if you're using this file, can we remove it?

File: .tests/models/nyc_taxi/model/serving_test.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  outputs:
+    pickups_next_hour:
+      profile: numerical
+      shape: scalar
+      type: int32
+install-command: pip install -r requirements.txt
+kind: Model
+monitoring:
+- config:
+    monitoring-model: nyc_taxi_monitoring_test:13
```
Comment (tidylobster) [inline .tests/models/nyc_taxi/model/serving_test.yaml:18]: Same here with a specific monitoring model. Omit if you can.

File: .tests/models/nyc_taxi/monitoring_model/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+/iforest.joblib
```
Comment (tidylobster) [inline .tests/models/nyc_taxi/monitoring_model/.gitignore:1]: I don't see any dvc files, I think we can safely remove this.

File: .tests/models/nyc_taxi/serving.yaml [hunk 9]
```diff
@@ -0,0 +1,65 @@
+kind: Model
+name: nyc_taxi_monitoring
+runtime: "hydrosphere/serving-runtime-python-3.6:0.1.2-rc0"
+install-command: "pip install -r requirements.txt"
```
Comment (tidylobster) [inline .tests/models/nyc_taxi/serving.yaml:4]: An old version of the runtime.

File: .tests/models/nyc_taxi/serving.yaml [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    with-health: true
+    config:
+      application: "nyc_taxi_monitoring"
+      threshold: "0"
+      operator: ">="
+---
+kind: Application
+name: "nyc_taxi"
+singular:
+  model: nyc_taxi:1
```
Comment (tidylobster) [inline .tests/models/nyc_taxi/serving.yaml:65]: Can we remove this?

File: .tests/utils.py [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+import requests
+from config import *
+from hydrosdk import LocalModel, DockerImage
+from hydroserving.core.model.parser import parse_model
+from hydroserving.util.yamlutil import yaml_file
+
+CLUSTER_URL = os.getenv("CLUSTER_URL", 'http://localhost:80')
+
+
+class ThresholdCmpOp(Enum):
```
Comment (tidylobster) [inline .tests/utils.py:15]: Why won't you use a similar class from hydrsdk?

File: .tests/utils.py [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+    return model
+
+
+def get_training_data_path(modelversion_id):
+    response = requests.get(f'{CLUSTER_URL}/monitoring/training_data?modelVersionId={modelversion_id}')
+    training_data_s3 = json.loads(response.text)
+    return training_data_s3
+
+
+def local_monitoring_model(model):
```
Comment (tidylobster) [inline .tests/utils.py:52]: I don't think I've seen this method being used anywhere. Is this a fixture?

**Review by tidylobster — CHANGES_REQUESTED** [review]
Great job overall! Requested minor changes.

**Comment by tidylobster** [comment]
Another minor point — we don't need to use huge training sets for test cases. We can shrink csv files to 5-10 lines, I think.

File: .tests/conftest.py [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+        .build()
+    app.lock_while_starting(TIMEOUT_SEC)
+    return app
+
+
+@pytest.fixture(scope="session", params=TEST_MODELS)
+def simulated_requests(cluster: Cluster, application: Application, request):
+    model = request.param
+    predictor = application.predictor()
+    sent_requests = simulate_traffic(model, predictor, demo_length=DATA_SAMPLE_SIZE)
```
[the remark this answers is not in the document]
Comment (BeardedWhale) [reply .tests/conftest.py:148]: Can you clarify this statement?

**Review by KineticCookie — CHANGES_REQUESTED** [review]
reject?

**Pull request closed without merging**